### ISSUE TITLE
Custom 404 page

### ISSUE THREAD

**Issue by HonkingGoose** [octoclairvoyant/octoclairvoyant-webapp#258]
### What feature would you like to have?

I just noticed our 404 page is really generic, and doesn't fit with our branding.
I'd like to at least customize the font/text style in the 404 page.

Even better would be to have use our mascot Octoclairvoyant with a suitable text under it:

"Octoclairvoyant cannot divine this page" or something like that. A bit of fun maybe. 😉 

Even better would be a nice custom pose/expression on the Octoclairvoyant macot, but that's probably out of scope.

### What problem will this feature solve?

Ensure 404 page is thematically the same as the rest of our pages.
Inject some personality into the page.

### Do you want to help make the feature?

Yes

### Is there anything else we need to know?

I don't how how hard/easy is it to customize the 404 page.
Next.js does support making a custom 404 page: https://nextjs.org/docs/advanced-features/custom-error-page#404-page

We need to discuss what kind of scope we want for the 404 page first!
I'm willing to design a new 404 page based on the scope we decide on together. 😉

**Comment by Belco90** [comment]
> Even better would be to have use our mascot Octoclairvoyant with a suitable text under it:
> 
> "Octoclairvoyant cannot divine this page" or something like that. A bit of fun maybe. 😉

I love this idea!

> Even better would be a nice custom pose/expression on the Octoclairvoyant macot, but that's probably out of scope.

This seems out of the scope indeed, but I have the original Adobe Illustrator file. Could we do something with that?

> I don't how how hard/easy is it to customize the 404 page.
> Next.js does support making a custom 404 page: https://nextjs.org/docs/advanced-features/custom-error-page#404-page
> 
> We need to discuss what kind of scope we want for the 404 page first!
> I'm willing to design a new 404 page based on the scope we decide on together. 😉

404 pages are really easy to set up in Next.js, it's just a matter to include a `404.ts` within `pages` as you saw in the documentation.

Regarding the scope for 404, I'd say that's pretty clear right? So we render it when accessing pages other than `/` or `/comparator`.

**Comment by HonkingGoose** [comment]
> 404 pages are really easy to set up in Next.js, it's just a matter to include a `404.ts` within `pages` as you saw in the documentation.

I bashed something together myself on a throwaway branch. Making the 404 page itself was quite easy, I just copy pasted most of the stuff we already use for our homepage right now. That got me most of the way there with regards to text/layout.

The biggest problem I have is that I couldn't get my "return to homepage" button to look `1:1` to the Figma design... 😞 
Otherwise things looked "okayish" to start with.

> This seems out of the scope indeed, but I have the original Adobe Illustrator file. Could we do something with that?

I guess the only thing we could change would be the text under the mascot? I'm not sure if changing the text is worth the effort to have a separate image. As the text can just be next/below the image in plain text anyways.

> Regarding the scope for 404, I'd say that's pretty clear right? So we render it when accessing pages other than `/` or `/comparator`.

That's fine!

What I meant with the scope was really more like: "how complicated do we want to make our custom 404 page?".

Proposed layout top to bottom:

1. Header
1. Centered image of mascot
1. Centered `h1` text below mascot image: "Cannot find page"
1. Left-aligned text "Octoclairvoyant cannot divine the page you're looking for."
1. Left-aligned `Button` to "Go to homepage" according to Figma design (`callToActionButton` component!)
1. Footer

If you want I can cobble this together in Figma first?

**Comment by Belco90** [comment]
> Proposed layout top to bottom:
> 
>     1. Header
> 
>     2. Centered image of mascot
> 
>     3. Centered `h1` text below mascot image: "Cannot find page"
> 
>     4. Left-aligned text "Octoclairvoyant cannot divine the page you're looking for."
> 
>     5. Left-aligned `Button` to "Go to homepage" according to Figma design (`callToActionButton` component!)
> 
>     6. Footer
> 
> 
> If you want I can cobble this together in Figma first?

Aah, I got it now. This looks ok for me! We have the mascot logo without text, so we can use that version here.

Thanks!

**Comment by HonkingGoose** [comment]
> We have the mascot logo without text, so we can use that version here.

Let's use the mascot logo without the text, that will look better!

---

I think we need proper components for our "call to action" button first. We will use the same style of button on the homepage and on the 404 page. That way we're not introducing changes between the button on the homepage and the one on the 404 page. 😄 

Do you want a new issue to track the work on the custom call to action button?

**Comment by Belco90** [comment]
Yeah, I think we can extract the call to action to its own component first!

**Comment by HonkingGoose** [comment]
Marking this `status:blocked` as it's a good idea to have a component for our call to action style button. So let's wait until #269 is done before starting work on this issue.
